server/routes/students.js

```diff
@@ -151,4 +153,24 @@ router.patch('/interests/add/:id', async (req, res, next) => {
     }
 })
 
+router.patch('/college/update/:id', passport.authenticate('jwt', {session: false}), async (req, res, next) => {
+    try {
+        // find or create College
+        // const collegeCountry = req.body.collegeCountry
+        let existingCollegeId = req.body.existingCollegeId
+        var college
+        if (existingCollegeId) {
+            college = await collegeSchema.findOne({_id: existingCollegeId})
+        }
+        let student = await studentSchema.findOne({_id: req.params.id})
+        student.collegeShortlist.push(college._id);
```

Comment:
This seems like unnecessary duplication - why check for an existing college, fetch the existing college, and then push the id of the fetched college to the shortlist when you have the id already?